pkgs/games/minecraft/default.nix

```diff
@@ -1,50 +1,95 @@
-{ stdenv, fetchurl, makeDesktopItem, makeWrapper
-, jdk, jre, libpulseaudio, libXxf86vm
+{ stdenv, fetchurl, dpkg, makeWrapper, jre, xlibs, zlib, nss, nspr, fontconfig, gnome2, cairo, expat, alsaLib, cups, dbus, atk, gtk2-x11, gdk_pixbuf, glib, curl, flite, freetype, libpulseaudio, systemd, libXxf86vm
 }:
 
-let
-  desktopItem = makeDesktopItem {
-    name = "minecraft";
-    exec = "minecraft";
-    icon = "minecraft";
-    comment = "A sandbox-building game";
-    desktopName = "Minecraft";
-    genericName = "minecraft";
-    categories = "Game;";
-  };
-
-  libPath = stdenv.lib.makeLibraryPath [
-    libpulseaudio
-    libXxf86vm # Needed only for versions <1.13
-  ];
+ stdenv.mkDerivation rec {
+  pname = "minecraft-launcher";
 
-in stdenv.mkDerivation {
-  name = "minecraft-2015-07-24";
+  version = "2.1.5965";
 
   src = fetchurl {
-    url = "https://s3.amazonaws.com/Minecraft.Download/launcher/Minecraft.jar";
-    sha256 = "04pj4l5q0a64jncm2kk45r7nxnxa2z9n110dcxbbahdi6wk0png8";
+    url = "https://launcher.mojang.com/download/Minecraft.deb";
+    sha256 = "1n559ywicg72ia1szmncifpfvjbvrnw1c8b477zmyjmfcd1lirbn";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ dpkg makeWrapper ];
+
+  sourceRoot = ".";
+
+  unpackPhase = "dpkg-deb -x $src .";
 
-  unpackPhase = "${jdk}/bin/jar xf $src favicon.png";
+  dontConfigure = true;
+  dontBuild = true;
 
-  installPhase = ''
-    mkdir -p $out/bin $out/share/minecraft
+  installPhase = let
+    envLibs = stdenv.lib.makeLibraryPath [
+      curl
+      flite
+      libpulseaudio
+      systemd
+      libXxf86vm # needed only for versions <1.13
+    ];
+  in ''
+    mkdir -p $out/bin
 
-    makeWrapper ${jre}/bin/java $out/bin/minecraft \
-      --add-flags "-jar $out/share/minecraft/minecraft.jar" \
-      --suffix LD_LIBRARY_PATH : ${libPath}
+    mv opt $out/opt
+
+    mv usr/share $out/share
+
+    substituteInPlace $out/share/applications/minecraft-launcher.desktop --replace /opt $out/opt
+
+    makeWrapper $out/opt/minecraft-launcher/minecraft-launcher $out/bin/minecraft-launcher --prefix LD_LIBRARY_PATH : ${envLibs} --prefix PATH : ${stdenv.lib.makeBinPath [ jre ]}
+  '';
 
-    cp $src $out/share/minecraft/minecraft.jar
-    cp -r ${desktopItem}/share/applications $out/share
-    install -D favicon.png $out/share/icons/hicolor/32x32/apps/minecraft.png
+  preFixup = let
+    libPath = stdenv.lib.makeLibraryPath ([
+      alsaLib
+      atk
+      cairo
+      cups
+      dbus
+      expat
+      fontconfig
+      freetype
+      gdk_pixbuf
+      glib
+      gnome2.GConf
+      gnome2.pango
+      gtk2-x11
+      nspr
+      nss
+      stdenv.cc.cc
+      zlib
+    ] ++
+    (with xlibs; [
```

Comment:
Please don't use `xlibs` it's a very old alias.
```
  xlibs = xorg; # added 2015-09
```
> aliases.nix

Reply:
Ah, that's why I couldn't find it in `all-packages.nix`. I wonder why `nix-locate` chose to display this one...